ci-operator/step-registry/mirror-images/by-oc-mirror/mirror-images-by-oc-mirror-commands.sh

```diff
@@ -98,53 +91,48 @@ extract_oc_mirror "${new_pull_secret}" "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRI
 oc_mirror_bin="${oc_mirror_dir}/oc-mirror"
 run_command "'${oc_mirror_bin}' version --output=yaml"
 
-# pre-check the expected image exists in official channel
-if "${oc_mirror_bin}" list releases --filter-by-archs ${ocp_image_arch} --channel=${channel_name} | grep -q "^${readable_version}$"; then
-    echo "${readable_version} is found in ${channel_name} on ${ocp_image_arch} arch"
-else
-    echo "${readable_version} is NOT found in ${channel_name} on ${ocp_image_arch} arch, exit..."
-    exit 1
-fi
 
 # set the imagesetconfigure
 image_set_config="image_set_config.yaml"
 cat <<END | tee "${image_set_config}"
 kind: ImageSetConfiguration
-apiVersion: mirror.openshift.io/v1alpha2
+apiVersion: mirror.openshift.io/v2alpha1
 mirror:
   platform:
-    architectures:
-      - "${ocp_image_arch}"
-    channels:
-      - name: ${channel_name}
-        minVersion: ${readable_version}
-        maxVersion: ${readable_version}
+    release: ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}
 END
 
 # https://github.com/openshift/oc-mirror/blob/main/docs/usage.md#authentication
 # oc-mirror only respect ~/.docker/config.json -> ${XDG_RUNTIME_DIR}/containers/auth.json
 mkdir -p "${XDG_RUNTIME_DIR}/containers/"
 cp -rf "${new_pull_secret}" "${XDG_RUNTIME_DIR}/containers/auth.json"
 
+unset REGISTRY_AUTH_PREFERENCE
+
 # execute the oc-mirror command
-cmd="'${oc_mirror_bin}' -c ${image_set_config} docker://${target_release_image_repo} --dest-skip-tls"
+cmd="'${oc_mirror_bin}' -c ${image_set_config} docker://${target_release_image_repo} --dest-tls-verify=false --v2 --workspace file://${mirror_output}"
```

Comment:
The output will be saved in `${mirror_output}` directly ?

Reply:
`${mirror_output}` was designed to save mirror output file, in your case, sounds like here should use `${oc_mirror_dir}`, if we do not set `--workspace` , it will use the current working dir, right?

Reply:
must set `--workspace` , it's necessary , and will save the mirror output file to there.

Reply:
${mirror_output} was designed to capture the whole output when the command is running, so that we can oc-mirror dynamic-generated folder to located the files that we needed.

I do not think `--workspace` parameter should use `${mirror_output}` here.